### DISCUSSION TITLE
Initialize preline JS components in a page displayed with ajax

### DISCUSSION THREAD

**Discussion by lethunder** [htmlstreamofficial/preline#208] [Q&A]
Hello guys,
all preline js components like tabs inside a page i hide and show with ajax and append to DOM are no more working. How do i initialize preline in jquery on ajax:success

Exemple code
$(document).on('ajax:success', ".display-in-offcanvas-js", function (event) {

var detail = event.detail;

var data = detail[0].body.innerHTML, status = detail[1], xhr = detail[2];

$(".application-offcanvas-js .content").html(data);

});
 
data contains html with preline js components

**Comment by dohahelmy** [comment]
any update fixing the issue?

**Comment by gongl** [comment]
I need this feature too. 

Please help.

**Comment by jahaganiev** [comment]
Hey folks, you may check out new Preline JavaScript page here with some explanation and AJAX example https://preline.co/docs/preline-javascript.html

Cheers!
